crates/router/src/connector/nuvei/transformers.rs

```diff
@@ -0,0 +1,619 @@
+            }
+            NuveiTransactionStatus::Processing => enums::AttemptStatus::Pending,
+        },
+        None => match response.status {
+            NuveiPaymentStatus::Failed | NuveiPaymentStatus::Error => enums::AttemptStatus::Failure,
+            _ => enums::AttemptStatus::Pending,
+        },
+    }
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, NuveiPaymentsResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, NuveiPaymentsResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            status: get_payment_status(&item.response),
+            response: match item.response.status {
+                NuveiPaymentStatus::Error => Err(types::ErrorResponse {
+                    code: item
+                        .response
+                        .err_code
+                        .ok_or(errors::ParsingError)?
+                        .to_string(),
+                    message: item.response.reason.clone().ok_or(errors::ParsingError)?,
+                    reason: None,
+                    status_code: item.http_code,
+                }),
+                _ => Ok(types::PaymentsResponseData::TransactionResponse {
+                    resource_id: types::ResponseId::ConnectorTransactionId(
+                        item.response.transaction_id.ok_or(errors::ParsingError)?,
+                    ),
+                    redirection_data: None,
+                    redirect: false,
+                    mandate_reference: None,
+                    connector_metadata: Some(
+                        serde_json::to_value(NuveiMeta {
+                            session_token: item.response.session_token.unwrap_or_default(),
+                        })
+                        .into_report()
+                        .change_context(errors::ParsingError)?,
+                    ),
+                }),
+            },
+            ..item.data
+        })
+    }
+}
+
+impl From<NuveiTransactionStatus> for enums::RefundStatus {
+    fn from(item: NuveiTransactionStatus) -> Self {
+        match item {
+            NuveiTransactionStatus::Approved => Self::Success,
+            NuveiTransactionStatus::Declined | NuveiTransactionStatus::Error => Self::Failure,
+            NuveiTransactionStatus::Processing => Self::Pending,
+        }
+    }
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::Execute, NuveiPaymentsResponse>>
+    for types::RefundsRouterData<api::Execute>
+{
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::Execute, NuveiPaymentsResponse>,
+    ) -> Result<Self, Self::Error> {
+        let refund_status = item
+            .response
+            .transaction_status
+            .map(|a| a.into())
```

Comment:
Try to avoid into whenever possible, pass the `From` function pointer directly to map function, this will help in jumping to the function definition wherever the transformation is happening.